helix-core/src/main/java/org/apache/helix/model/IdealState.java

```diff
@@ -536,4 +574,16 @@ public boolean isEnabled() {
   public void enable(boolean enabled) {
     _record.setSimpleField(IdealStateProperty.HELIX_ENABLED.name(), Boolean.toString(enabled));
   }
+
+  /**
+   * Get the mangled IdealState name if resourceGroup is enable.
+   *
+   * @param resourceName
+   * @param resourceTag
+   *
+   * @return
+   */
+  public static String getIdealStateName(String resourceName, String resourceTag) {
```

Comment:
why do we need this method? This convention can be completely handled on client side rt?